### ISSUE TITLE
Stderr 

### ISSUE THREAD

**Issue by SephVelut** [rust-lang/cargo#5641]
I don't get the justification #1473 for switching stdout with stderr. If you want special formatting from stdout then its on you to sed or awk. 

From an abstract perspective they are both just file descriptors, but there is no reason to trickle down a formatting use case to how stderr and stdout are conventionally treated by everyone else. That is, write to stderr if the command exited with a code other than 0.

**Comment by matklad** [comment]
@SephVelut I don't quite understand what problem exactly are you trying to solve? Could you rephrase your question? 

The current strategy for output in Cargo is that all human-reaadble things go to stderr, and all machine readable things go to stdout. That way, you can safely parse Cargo's stdout without accounting for various diagnostic messages.

**Comment by SephVelut** [comment]
Shouldn't all output go to stdout and reserve stderr for errors in executing the program? Most command line programs work this way. Stderr and stdout can both be piped down at the same time to other commands and stderr can be treated like a failure case. Not of the program but of the command. There's a distinction. Regardless there is no distinction between machine readable and human readable. Unless it's literally binary, that's purely subjective.

Parsing Cargo's output should be the responsibility of the user.

**Comment by alexcrichton** [comment]
This is currently an intentional design decision where you can execute commands like `cargo run | foo` where `foo` only receives the output of the program run, nothing from Cargo and/or rustc.

**Comment by SephVelut** [comment]
Why not just wait for the OP to respond/close? What's the hurry?

I don't understand the design decision, can you please explain more? Wouldn't it be weird if a function that returned `Result` had logical or computational values inside Err, instead of a diagnostic of the error that occurred?

**Comment by alexcrichton** [comment]
The design decision here is for all information output of Cargo to go to one stream, stderr. Sort of how all compiler error messages go to stderr as well. The thinking is then that stdout is reserved for programs Cargo runs, so Cargo never produces meaningful output on stdout. That way you can throw away stderr from Cargo or only capture stdout and you're guaranteed to get your program's output
